packages/app-desktop/gui/NoteListControls/NoteListControls.tsx

```diff
@@ -69,6 +71,73 @@ const SortOrderButtonsContainer = styled.div`
 
 function NoteListControls(props: Props) {
 	const searchBarRef = useRef(null);
+	const newNoteRef = useRef(null);
+	const newTodoRef = useRef(null);
+	const noteControlsRef = useRef(null);
+	const searchAndSortRef = useRef(null);
+
+	const [noteBtnText, setNoteBtnText] = useState('New note');
+	const [todoBtnText, setTodoBtnText] = useState('New to-do');
+	const [breakpoint, setBreakpoint] = useState('l');
+	const [iconNote, setIconNote] = useState('fas fa-plus');
+	const [iconTodo, setIconTodo] = useState('fas fa-plus');
+
+	useEffect(() => {
+		const breakpoints = [{ s: 135 }, { md: 189 }, { l: 470 }, { xl: 500 }];
+		// Find largest breakpoint that width is less than
+		const index = breakpoints.map(x => Object.values(x)[0])
+			.findIndex(x => props.width < x);
+
+		const bp = index === -1 ? Object.keys(breakpoints[breakpoints.length - 1])[0] : Object.keys(breakpoints[index])[0];
+
+		setBreakpoint(bp);
+	}, [props.width]);
+
+	useEffect(() => {
+		if (breakpoint === 's') {
+			setNoteBtnText('');
+			setTodoBtnText('');
+		} else if (breakpoint === 'md') {
+			setNoteBtnText('note');
+			setTodoBtnText('to-do');
+		} else {
+			setNoteBtnText('New note');
+			setTodoBtnText('New to-do');
+		}
+	}, [breakpoint]);
+
+	useEffect(() => {
+		if (breakpoint === 's') {
+			newNoteRef.current.style.padding = '0px 18px 0px 18px';
+			newTodoRef.current.style.padding = '0px 18px 0px 18px';
+		} else {
+			newNoteRef.current.style.padding = '0px 4px 0px 4px';
+			newTodoRef.current.style.padding = '0px 4px 0px 4px';
+		}
+
+		if (breakpoint === 'xl') {
+			noteControlsRef.current.style.flexDirection = 'row';
+			searchAndSortRef.current.style.flex = '2 1 auto';
+		} else {
+			noteControlsRef.current.style.flexDirection = 'column';
+		}
+	}, [breakpoint]);
+
+	useEffect(() => {
+		if (breakpoint === 's') {
+			setIconNote('icon-note');
+			setIconTodo('far fa-check-square');
+		} else if (breakpoint === 'md') {
+			setIconNote('fas fa-plus');
+			setIconTodo('fas fa-plus');
+		} else if (breakpoint === 'l') {
+			setIconNote('fas fa-plus');
+			setIconTodo('fas fa-plus');
+		} else {
+			setIconNote('fas fa-plus');
+			setIconTodo('fas fa-plus');
+		}
+	}, [breakpoint]);
```

Comment:
In fact it looks like all these effects are to calculate derived values, so you could use `useMemo` for all of this. `useEffect` is generic and indeed can be used to replace all other hooks, but it's better to choose more specific ones to show intent and, in this case, to reduce the number of state variables

Reply:
Same here. Changed and let me know if this is what you had in mind.